app/src/main/java/org/traccar/client/StatusWidget.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 - 2022 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.client
+
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.*
+import android.widget.RemoteViews
+
+import androidx.preference.PreferenceManager
+
+/**
+ * Implementation of Status Widget functionality, which can display the status of the service -
+ * whether it is enabled or not.
+ */
+class StatusWidget : AppWidgetProvider() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        if (TrackingService.ACTION_STARTED == intent.action
+                || TrackingService.ACTION_STOPPED == intent.action) {
+            updateWidgets(context)
+        }
+        else
+            super.onReceive(context, intent)
```

Comment:
Do we need else?

Reply:
Yes, the base class should handle widget-specific notifications like `ACTION_APPWIDGET_UPDATE` and alike.